src/framework/components/rigid-body/system.js

```diff
@@ -552,6 +560,240 @@ class RigidBodyComponentSystem extends ComponentSystem {
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {number} radius - The radius of the cylinder.
+     * @param {number} height - The total height of the cylinder from tip to tip.
+     * @param {number} axis - The local space axis with which the cylinder's length is aligned. 0 for X, 1 for Y and 2 for Z. Defaults to 1 (Y-axis).
+     * @param {Vec3} position - The world space position for the cylinder to be.
+     * @param {Vec3} rotation - The world space rotation for the cylinder to have.
+     *
+     * @returns {RaycastResult[]} An array of cylindercast hit results (0 length if there were no hits).
+     */
+    cylinderCast(radius, height, axis, position, rotation) {
+        Debug.assert(Ammo.ConcreteContactResultCallback, 'pc.RigidBodyComponentSystem#cylinderCast: Your version of ammo.js does not expose Ammo.ConcreteContactResultCallback. Update it to latest.');
+
+        if (axis === 0) {
+            axis = 'btCylinderShapeX';
+        } else if (axis === 2) {
+            axis = 'btCylinderShapeZ';
+        } else {
+            axis = 'btCylinderShape';
+        }
+
+        return this._shapecast(new Ammo[axis](radius, height), position, rotation);
+    }
+
+    /**
+     * Perform a collision check on the world and return all entities the sphere hits.
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {number} radius - The radius of the sphere.
+     * @param {Vec3} position - The world space position for the sphere to be.
+     * @param {Vec3} rotation - The world space rotation for the sphere to have.
+     *
+     * @returns {RaycastResult[]} An array of spherecast hit results (0 length if there were no hits).
+     */
+    sphereCast(radius, position, rotation) {
+        Debug.assert(Ammo.ConcreteContactResultCallback, 'pc.RigidBodyComponentSystem#sphereCast: Your version of ammo.js does not expose Ammo.ConcreteContactResultCallback. Update it to latest.');
+
+        return this._shapecast(new Ammo.btSphereShape(radius), position, rotation);
+    }
+
+    /**
+     * Perform a collision check on the world and return all entities the shape hits.
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {Ammo.btCollisionShape} shape - The Ammo.btCollisionShape to use for collision check.
+     * @param {Vec3} position - The world space position for the shape to be.
+     * @param {Vec3} rotation - The world space rotation for the shape to have.
+     *
+     * @returns {RaycastResult[]} An array of spherecast hit results (0 length if there were no hits).
+     * @private
+     */
+    _shapecast(shape, position, rotation) {
```

Comment:
Am I right in saying the `shape` instance is not currently being destroyed anywhere?

Reply:
Yes it is not destroyed. I don't have full knowledge over Ammo.js so I don't know if it will affect the btRigidBody if destroyed as the btRigidBody is never destroyed to avoid re-creating one every call (and is only created when calling it for first time).

Reply:
I'm not 100% sure what the best solution is here, but at the moment, the code would definitely leak the created shape on every cast. How about doing:

```
shapecastBody.setCollisionShape(null);
Ammo.destroy(shape);
```

Reply:
I just tested out and it works. Just pushed a new commit to fix it.

Reply:
Nice, thanks.